% Chapter discussion

\chapter{Abstract} %-1pagina}
    
 % Main chapter title

\label{Chapter0} % For referencing the chapter elsewhere, use \ref{Chapter4} 

%----------------------------------------------------------------------------------------

% Define some commands to keep the formatting separated from the content 
\newcommand{\keyword}[1]{\textbf{#1}}
\newcommand{\tabhead}[1]{\textbf{#1}}
\newcommand{\code}[1]{\texttt{#1}}
\newcommand{\file}[1]{\texttt{\bfseries#1}}
\newcommand{\option}[1]{\texttt{\itshape#1}}

%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Studies of genomes typically assume a single linear reference genome. This can make it difficult to observe sequences in other genomes that are divergent from the reference, which in turn can limit the accuracy and completeness of our analyses. To address this limitation, we can work with pangenome reference systems that represent the mutual relationships between many complete genomes without using a single one as a point of reference. These models often encode genomes and their mutual alignment in a compact graphical structure. Here, I begin the development of a software library for the statistical analysis of population genetic parameters of genomes stored in these graphical pangenomic data models. Because a pangenome embeds the linear genomes from which it is constructed, we can choose a particular reference genome and project the variants on it(which appear as bubbles in the graph). Therefore, I focused on develop algorithms for bubble detection, generating Variant Call Format (VCF) files directly from graphs. I then developed methods to annotate these variants with allele frequencies of genomes embedded in the input graph, and then explored the calculation of Fst. % for three different time points.

%\noindent
%We can use this variants projection to drive standard population genetic analyses. In my thesis, I focused on the calculation of Fst for three different time.

