% Chapter discussion

\chapter{Aim of the thesis} %-1,2 pagine} % Main chapter title

\label{Chapter4} % For referencing the chapter elsewhere, use \ref{Chapter4} 

%----------------------------------------------------------------------------------------

%Illustra in modo chiaro e coinciso le finalit√† del lavoro sperimentale svolto dal candidato. 

We are updating our methods to study genomic sequences to meet new challenges and reach for new possibilities offered by contemporary sequencing technology. The development of high-quality long-read sequencing methods allow us to routinely assemble whole genomes \emph{de novo}. This situation allows us to consider whole genomes in our analyses, allowing us to expose the full range of genomic variation to downstream studies. However, doing so requires updates to our methods, to allow us to consider the mutual relationships between a collection of sequences in a complete way.\\

Pangenomes are large, and unwieldy to work with as raw collections of sequences. One possible approach to processing them considers the collection of genomes and their mutual alignment in a compact, graphical model. As a lossless representation of the pangenome and its embedded sequence variation, these \emph{variation graphs} should in principle support any kind of population genetic analysis that would be completed on simpler representations of the genomes and their variation. But, because this pangenomic approach is quite recent, the software for population genetic analyses currently available are still mostly based on genomic data in the linear format.\\

In this thesis, I respond to this need by implementing a set of tools for population genetics on genome graphs. At a high level, my work has two parts. I first uncover genomic variation embedded in pangenomic variation graphs by developing and implementing straightforward algorithms for \emph{bubble} detection on variation graphs. I then demonstrate the calculation of basic population genetic parameters over these variations, considering the set of genomes embedded in the graph.\\

Specifically, this work required the completion of a series of tasks:
\begin{itemize}
    \item To write code for parsing variable regions of the genomic data in graphical fragment assembly format; 
    \item To write code to calculate standard summary statistics for population genomic analysis; 
    \item To test the functions of the library on simulated data;
    \item To test the functions of the library on a case study.
\end{itemize}




